scripts/process.py

```diff
@@ -0,0 +1,161 @@
+        ["marketcap", "Market Cap"],
+        ["EBITDA", "EBITDA"],
+        ["priceToSales", "Price/Sales"],
+        ["priceToBook", "Price/Book"]
+    ]
+})
+
+
+def rename(package: PackageWrapper):
+    package.pkg.descriptor['resources'][0]['name'] = 's-p-constituents'
+    package.pkg.descriptor['resources'][0]['path'] = 'data/constituents.csv'
+    package.pkg.descriptor['resources'][1]['name'] = 'constiuents-financials'
+    package.pkg.descriptor['resources'][1]['path'] = 'data/constiuents-financials.csv'
+    yield package.pkg
+    res_iter = iter(package)
+    first: ResourceWrapper = next(res_iter)
+    second: ResourceWrapper = next(res_iter)
+    yield first.it
+    yield second.it
+    yield from package
+
+
+def extract_constituents(html):
+    datadir = 'data'
+    if not exists(datadir):
+        mkdir(datadir)
+    soup = BeautifulSoup(html, 'html.parser')
+    table = soup.find("table", {"class": "wikitable sortable"})
+    # Fail now if we haven't found the right table
+    header = table.findAll('th')
+    if header[0].string != "Ticker symbol" or header[1].string != "Security":
+        raise Exception("Can't parse wikipedia's table!")
+
+    # Retreive the values in the table
+    rows = table.findAll('tr')
+    for row in rows:
+        fields = row.findAll('td')
+        if fields:
+            yield {
+                'Symbol': fields[0].text,
+                'Name': fields[1].text,
+                'Sector': fields[3].text
+            }
+
+
+# helper function to append column data for each iex type and field pair
+def append_column_data(rows, row_number, data_dict):
+    for iex_type, field in IEX_TYPES_FIELDNAMES_TITLES.items():
+        for field_name in field:
+            rows[row_number].append(data_dict[iex_type][field_name[0]])
+
+
+def extract_financial_data():
+    # get constituents data
+    reader = csv.reader(open('data/constituents.csv'))
+
+    # gather field titles and IEX field names
+    iex_fieldnames = []
+    field_titles = []
+    for iex_type in sorted(IEX_TYPES_FIELDNAMES_TITLES.values()):
+        iex_fieldnames += [field[0] for field in iex_type]
+        field_titles += [field[1] for field in iex_type]
+
+    outrows = [row for row in reader]
+    # add new column titles to outfile
+    outrows[0] += field_titles + ['SEC Filings']
+    # gather symbols
+    symbols = [row[0] for row in outrows[1:]]
+    # gather IEX "types"
+    iex_types = IEX_TYPES_FIELDNAMES_TITLES.keys()
+    # build first part of url, IEX uses comma delimiters between values
+    url = IEX_BASE_URL + '?'
+    url += 'types=' + ','.join(iex_types)
+    url += '&filter=' + ','.join(iex_fieldnames)
+    url += '&symbols='
+
+    # make requests in chunks of symbols at a time
+    # store response data in outrows
+    for index in range(0, len(symbols), IEX_NUMBER_OF_SYMBOLS_PER_REQUEST):
+        query = url + ','.join(symbols[index:index + IEX_NUMBER_OF_SYMBOLS_PER_REQUEST])
+        resp = urllib.request.urlopen(query)
```

Comment:
why not use `reqests.get()` liberary and it's `json` method?
```
resp = requests.get(query)
results = resp.json()
```

Reply:
FIXED